crates/languages/src/rust.rs

```diff
@@ -389,6 +389,22 @@ impl ContextProvider for RustContextProvider {
                 tags: vec!["rust-test".to_owned()],
                 ..TaskTemplate::default()
             },
+            TaskTemplate {
+                label: format!(
+                    "cargo test -p {} {}",
+                    RUST_PACKAGE_TASK_VARIABLE.template_value(),
+                    VariableName::Filename.template_value(),
+                ),
+                command: "cargo".into(),
+                args: vec![
+                    "test".into(),
+                    "-p".into(),
+                    RUST_PACKAGE_TASK_VARIABLE.template_value(),
+                    VariableName::Filename.template_value(),
+                ],
+                tags: vec!["rust-file-test".to_owned()],
```

Comment:
Can we use a different tag than `rust-file-test`? I want to introduce a tag for testing whole file if we detect at least one test in it and I guess I'd like to use a tag similar to the one you have there.
I wonder if we could even perhaps use `rust-test` again there.

Reply:
Unfortunately, we cannot reuse the `rust-test` tag because it will run the wrong task.
<img width="295" alt="Screenshot 2024-05-18 at 18 43 24" src="https://github.com/zed-industries/zed/assets/62463826/4dc308f2-4c9a-4991-972c-fdea16c151f7">

Reply:
Oh, right; you're totally right.